### PR TITLE
Small refactoring for null separated values

### PR DESCRIPTION
**Pull request by hhugo**
Introduce null_separated split and concat in misc and remove duplicated implementation from the codebases

### CODE REVIEW

**Review by smuenzel — APPROVED** [review]
This looks like a good improvement, especially removing some of the more complicated string logic.

File: utils/misc.mli [hunk 1]
```diff
@@ -257,6 +257,9 @@ val protect_writing_to_file
           channel. If the function raises an exception then [filename]
           will be removed. *)
 
+val concat_null_separated : string list -> string
+val split_null_separated : string -> string list
```
Comment (nojb) [inline utils/misc.mli]: A docstring is needed, especially explaining the fact that the result is NULL-terminated (which is different from the standard "concat" terminology in OCaml). In fact, a different name may be better: `concat_null_terminated`, `split_null_terminated`.

File: utils/misc.ml [hunk 2]
```diff
@@ -423,6 +423,17 @@ let find_first_mono =
...
+let split_null_separated s =
+  let[@tail_mod_cons] rec discard_last_sep = function
+    | [] | [""] -> []
+    | x :: xs -> x :: discard_last_sep xs
+  in
+  discard_last_sep (String.split_on_char '\000' s)
+
+let concat_null_separated = function
+  | [] -> ""
+  | l -> String.concat "\000" l ^ "\000"
```
Comment (gasche) [inline utils/misc.ml]: The use of `^` means that we copy the output twice, which is unnecessary and could be noticeable if the output is large. You could use `String.concat "\000" (l @ [""])` instead which only copies the list spine.

File: utils/misc.mli (hunk 1 above)
Comment (hhugo) [reply utils/misc.mli]: changed and documented

File: utils/misc.ml (hunk 2 above)
Comment (hhugo) [reply utils/misc.ml]: changed

**Comment by hhugo** [comment]
changelog probably unnecessary

File: utils/misc.mli [hunk 3]
```diff
@@ -257,6 +257,14 @@ val protect_writing_to_file
           channel. If the function raises an exception then [filename]
           will be removed. *)
 
+(** [concat_null_terminated [x1;x2; ... xn]] is
+    [x1 ^ "\000" ^ x2 ^ "\000" ^ ... ^ xn ^ "\000"] *)
+val concat_null_terminated : string list -> string
```
Comment (nojb) [inline utils/misc.mli]: The rest of the comments in this file are placed _after_ the corresponding value.

**Review by nojb — APPROVED** [review]
LGTM

**Comment by nojb** [comment]
Planning to (squash-)merge once the CI passes.

File: utils/misc.mli (hunk 3 above)
Comment (hhugo) [reply utils/misc.mli]: done

**Comment by nojb** [comment]
@hhugo you need to update the depend files (`make alldepend`).

**Pull request merged**